src/storage.rs

```diff
@@ -308,6 +317,23 @@ where
             _ => panic!("Storage should already have been stopped!"),
         }
     }
+
+    /// Removes elements up to the given [`idx`] from storage.
+    pub fn garbage_collect(&mut self, idx: u64) {
+        match self.sequence {
+            PaxosSequence::Active(ref mut s) => {
+                s.garbage_collect(idx - self.paxos_state.get_gc_idx());
+                self.paxos_state.set_gc_idx(idx);
+            }
+            PaxosSequence::Stopped(_) => {} // todo what to do when paxos is stopped?
```

Comment:
If we are stopped, we can carbage collect too. Here, I think we can actually use the OR `|` in pattern matching :D 

```suggestion
            PaxosSequence::Active(ref mut s) | PaxosSequence::Stopped(_) => {
                s.garbage_collect(idx - self.paxos_state.get_gc_idx());
                self.paxos_state.set_gc_idx(idx);
            }
```